examples/knative/python/kn-py-vro/handler.py

```diff
@@ -0,0 +1,325 @@
+            param['type'] = "string"
+            param['value'] = {"string": {"value": value}}
+
+    elif type(value) == bool:
+        param['type'] = "boolean"
+        param['value'] = {"boolean": {"value": value}}
+
+    elif type(value) == dict:  # a dict is likely a VC manged object reference
+        for v in value.values():
+            # search the items of the dict and look for contained dicts with a 'Type' property
+            if type(v) == dict and 'Type' in v and 'Value' in v:
+                param['type'] = f"VC:{v['Type']}"
+                param['value'] = {
+                    "sdk-object": {
+                        "type": param['type'],
+                        "id": f'{vchost},id:{v["Value"]}'
+                    }
+                }
+        if not 'type' in param: # if no vc managed object was found, pass the data as JSON
+            param['type'] = 'string'
+            param['value'] = {"string": {"value": json.dumps(value)}}
+
+    elif type(value) == list:
+        param['type'] = "Properties"
+        properties = []
+        for v in value:
+            if 'Key' in v and 'Value' in v:
+                # pass the key and value to this function recursively to get a correctly structured value
+                vroParam = getVroInputParam((v["Key"], v["Value"]))
+                properties.append({"key": v["Key"], "value": vroParam["value"]})
+        param['value'] = {"properties": {"property": properties}}
+
+    elif value is None:
+        param = None
+
+    else:  # Unhandled data type - try and wrangle to a string
+        app.logger.debug(f'{bgc.WARNING}Unhandled data type "{type(value).__name__}" - forcing to string{bgc.ENDC}')
+        param['type'] = "string"
+        param['value'] = {"string": {"value": str(value)}}
+
+    if type(param) == dict:  # set the parameter name for valid types
+        param["name"] = name[0].lower() + name[1:]
+    return param
+
+
+def get_response(message, code):
+    """
+    Get's a formated response object
+
+    Args:
+        message (string): The response message
+        code (integer): The result code of the response
+
+    Returns:
+        Object: correctly formatted response object
+    """
+    if 200 <= code < 300:
+        app.logger.info(message)
+    else:
+        app.logger.error(f'{bgc.FAIL}{message}{bgc.ENDC}')
+    msg = {
+        'status': code,
+        'message': message,
+    }
+    resp = jsonify(msg)
+    resp.status_code = code
+    return resp
+
+
+app = Flask(__name__)
+@app.route('/', methods=['POST'])
+def handle():
+    """
+    Handles a request to the function
+
+    Returns:
+        Object: HTTP response object
+    """
+    global vchost, bearer_token
+
+    # Load the Events that function gets from vCenter through the Event Router
+    app.logger.debug(f'{bgc.HEADER}Reading Cloud Event{bgc.ENDC}')
+    cevent = from_http(request.headers, request.get_data(),None)
```

Comment:
question: could this throw?

Reply:
Yes it could - and it has in my testing when I've done something stupid! I'll catch this.